test/unit/wifi.js

```diff
@@ -0,0 +1,236 @@
+    this.setNetworkPassword = sinon.spy(commands, 'setNetworkPassword');
+    this.commitWirelessCredentials = sinon.spy(commands, 'commitWirelessCredentials');
+    this.reconnectWifi = sinon.spy(commands, 'reconnectWifi');
+    this.tessel = TesselSimulator();
+
+    done();
+  },
+  tearDown: function(done) {
+    this.tessel.close();
+    this.connectToNetwork.restore();
+    this.logsWarn.restore();
+    this.logsInfo.restore();
+    this.setNetworkSSID.restore();
+    this.setNetworkPassword.restore();
+    this.commitWirelessCredentials.restore();
+    this.reconnectWifi.restore();
+    done();
+  },
+  noSSID: function(test) {
+    var self = this;
+    test.expect(3);
+    this.tessel.connectToNetwork({
+        ssid: undefined,
+        password: 'fish'
+      })
+      .catch(function(error) {
+        test.ok(error);
+        test.equal(self.setNetworkSSID.callCount, 0);
+        test.equal(self.setNetworkPassword.callCount, 0);
+        test.done();
+      });
+  },
+  noPassword: function(test) {
+    var self = this;
+    test.expect(3);
+    this.tessel.connectToNetwork({
+        ssid: 'tank',
+        password: undefined
+      })
+      .catch(function(error) {
+        test.ok(error);
+        test.equal(self.setNetworkSSID.callCount, 0);
+        test.equal(self.setNetworkPassword.callCount, 0);
+        test.done();
+      });
+  },
+  properCredentials: function(test) {
+    var self = this;
+    test.expect(6);
+    var creds = {
+      ssid: 'tank',
+      password: 'fish'
+    };
+    this.tessel.connectToNetwork(creds)
+      .then(function() {
+        test.equal(self.setNetworkSSID.callCount, 1);
+        test.equal(self.setNetworkPassword.callCount, 1);
+        test.equal(self.commitWirelessCredentials.callCount, 1);
+        test.equal(self.reconnectWifi.callCount, 1);
+        test.ok(self.setNetworkSSID.lastCall.calledWith(creds.ssid));
+        test.ok(self.setNetworkPassword.lastCall.calledWith(creds.password));
+        test.done();
+      })
+      .catch(function(error) {
+        test.fail(error);
+      });
+
+    // Test is expecting two closes...;
+    setImmediate(function() {
+      self.tessel._rps.emit('close');
+      setImmediate(function() {
+        self.tessel._rps.emit('close');
+      });
+    });
+  }
+};
+
+function TesselSimulator() {
+  var tessel = new Tessel();
+  tessel._rps = new RemoteProcessSimulator();
+  tessel.connection = {
+    exec: function(command, callback) {
+      if (typeof callback === 'function') {
+        callback(null, tessel._rps);
+      }
+    },
+    end: function(callback) {
+      if (typeof callback === 'function') {
+        callback();
+      }
+    }
+  };
+
+  return tessel;
+}
+
+function RemoteProcessSimulator() {
+  this.stdin = new stream.Writable();
+  this.stdout = new stream.Readable();
+  this.stderr = new stream.Readable();
+
+  this.stdout._read = function() {};
+  this.stderr._read = function() {};
+}
```

Comment:
If you think this will be useful for other test files, then move it out to a separate file that can be require()'ed as needed :)